### PR TITLE
Add build with java doc

### PR DESCRIPTION
**Pull request by raoufchebri**

### CODE REVIEW

File: docs/build-with-java.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+In this section, we will walk through and explain the code for the different commands.
+As explained in the Getting Started page, the project is structured as follow:
+
+-   Migrate (`com.carepet.Migrate`) - creates the `carepet` keyspace and tables
+-   Collar (`com.carepet.Sensor`) - generates a pet health data and pushes it into the storage
+-   Web app (`com.carepet.server.App`) - REST API service for tracking pets health state
+
+### Migrate
+
+The `./bin/migrate.sh --hosts $NODE1 --datacenter datacenter1` command executes the main function in the `Migrate` class located `Migrate.java`. The function creates the keyspace and tables that will be used by the collar and server services.
```
Comment (tzach) [inline docs/build-with-java.md:14]:
```suggestion
The `./bin/migrate.sh --hosts $NODE1 --datacenter datacenter1` command executes the main function in the `Migrate` class located `Migrate.java`. The function creates the keyspace and tables used by the collar and server services.
```

File: docs/build-with-java.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+	final Migrate client = new Migrate(config);
+	client.createKeyspace();
+	client.createSchema();
+	client.printMetadata();
+}
+```
+
+There is a lot to unpack here. Let's break the code down line by line:
+
+The `config` object parses the arguments passed in the migrate command. In our case it's `hosts` and `datacenter`. The `hosts` argument expect the IP address of one of the nodes. The `datacenter` argument is `datacenter1` by default but could be different if you are using Scylla Cloud for instance. The command also accepts `username` and `password` arguments if required.
```
Comment (tzach) [inline docs/build-with-java.md:31]:
```suggestion
The `config` object parses the arguments passed in the migrate command. In our case it's `hosts` and `datacenter`. The `hosts` argument expects the IP address of one of the nodes. The `datacenter` argument is `datacenter1` by default but could be different if you use Scylla Cloud. The command also accepts `username` and `password` arguments if required.
```

File: docs/build-with-java.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+In the care-pet example, you will use `$ curl http://127.0.0.1:8000/api/owner/{id}` and expect the following response:
+
+```
+
+[{"address":"home","age":57,"name":"tlmodylu","owner_id":"a05fd0df-0f97-4eec-a211-cad28a6e5360","pet_id":"a52adc4e-7cf4-47ca-b561-3ceec9382917","weight":5}]
+
+```
+
+The controller is defined in `ModelController.java`, and implements the GET methods to acces owners, pets and sensors data.
```
Comment (tzach) [inline docs/build-with-java.md:238]:
```suggestion
The controller is defined in `ModelController.java`, and implements the GET methods to access owners, pets and sensors data.
```

**Review by tzach — COMMENTED** [review]
Few minor grammar suggestions (from Grammarly)

File: docs/build-with-java.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
+## Build an IoT App with Java
+
+### Architecture
+
+In this section, we will walk through and explain the code for the different commands.
+As explained in the Getting Started page, the project is structured as follow:
+
+-   Migrate (`com.carepet.Migrate`) - creates the `carepet` keyspace and tables
```
Comment (annastuchlik) [inline docs/build-with-java.md:8]:
```suggestion
-   Migrate (`com.carepet.Migrate`) - Creates the `carepet` keyspace and tables.
```

File: docs/build-with-java.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+-   Collar (`com.carepet.Sensor`) - generates a pet health data and pushes it into the storage
```
Comment (annastuchlik) [inline docs/build-with-java.md:9]:
```suggestion
-   Collar (`com.carepet.Sensor`) - Generates pet health data and pushes it into the storage.
```

File: docs/build-with-java.md [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+-   Web app (`com.carepet.server.App`) - REST API service for tracking pets health state
```
Comment (annastuchlik) [inline docs/build-with-java.md:10]:
```suggestion
-   Web app (`com.carepet.server.App`) - REST API service for tracking pets' health state.
```

File: docs/build-with-java.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+### Migrate
+
+The `./bin/migrate.sh --hosts $NODE1 --datacenter datacenter1` command executes the main function in the `Migrate` class located `Migrate.java`. The function creates the keyspace and tables that will be used by the collar and server services.
+
+The below code in the `Migrate.java` file call the `createKeyspace` , `createSchema` then the `printMetadata` functions.
```
Comment (annastuchlik) [inline docs/build-with-java.md:16]:
```suggestion
The following code in the `Migrate.java` file calls the `createKeyspace` , `createSchema` , and `printMetadata` functions.
```

File: docs/build-with-java.md [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+	final Config config = Config.parse(new Config(), args);
+
+	final Migrate client = new Migrate(config);
+	client.createKeyspace();
+	client.createSchema();
+	client.printMetadata();
+}
+```
+
+There is a lot to unpack here. Let's break the code down line by line:
```
Comment (annastuchlik) [inline docs/build-with-java.md:29]:
```suggestion
Let's break down the code line by line:
```

File: docs/build-with-java.md (hunk 2 above)
Comment (annastuchlik) [reply docs/build-with-java.md:31]: A small update to Tzach's comment:
The `config` object parses the arguments passed in the `migrate command`. In this case, the arguments are `hosts` and `datacenter`. The `hosts` argument expects the IP address of one of the nodes. The `datacenter` argument is `datacenter1` by default but could be different if you use Scylla Cloud. The command also accepts `username` and `password` arguments if required.

File: docs/build-with-java.md [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+
+The `config` object parses the arguments passed in the migrate command. In our case it's `hosts` and `datacenter`. The `hosts` argument expect the IP address of one of the nodes. The `datacenter` argument is `datacenter1` by default but could be different if you are using Scylla Cloud for instance. The command also accepts `username` and `password` arguments if required.
+
+The `createKeyspace` function creates a new `CqlSession` then executes the following CQL query stored in the `resources/care-pet-keyspace.cql` file:
```
Comment (annastuchlik) [inline docs/build-with-java.md:33]:
```suggestion
The `createKeyspace` function creates a new `CqlSession`, then executes the following CQL query stored in the `resources/care-pet-keyspace.cql` file:
```

File: docs/build-with-java.md [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+	}
+}
+```
+
+```
+CREATE KEYSPACE IF NOT EXISTS carepet WITH replication = { 'class': 'NetworkTopologyStrategy', 'replication_factor': '3' } AND durable_writes = TRUE;
+```
+
+The CQL query above creates a new keyspace named carepet, with `NetworkTopologyStrategy` as replication strategy and a replication factor of 3.
+More information about keyspace and replication on [Scylla University](https://university.scylladb.com/courses/data-modeling/lessons/basic-data-modeling-2/topic/keyspace/).
```
Comment (annastuchlik) [inline docs/build-with-java.md:49]:
```suggestion
See [Scylla University](https://university.scylladb.com/courses/data-modeling/lessons/basic-data-modeling-2/topic/keyspace/) for more information about keyspaces and replication.
```

File: docs/build-with-java.md [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+	LOG.info("creating table...");
+	try (CqlSession session = keyspace()) {
+	    for (String cql : Config.getResource("care-pet-ddl.cql").split(";")) {
+		session.execute(cql);
+	    }
+	}
+}
+```
+
+As the name suggests, the `printMetadata` function will then print the metadata related to the `carepet` keyspace and confirm that the tables were properly created.
```
Comment (annastuchlik) [inline docs/build-with-java.md:70]:
```suggestion
The `printMetadata` function will print the metadata related to the `carepet` keyspace and confirm that the tables are properly created.
```

File: docs/build-with-java.md [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+AND max_index_interval = 2048
+AND memtable_flush_period_in_ms = 0
+AND min_index_interval = 128
+AND read_repair_chance = 0.0
+AND speculative_retry = '99.0PERCENTILE';
+```
+
+### Sensor
+
+The sensor service simulates the collar's activity. You can use the below command to run the sensor service:
```
Comment (annastuchlik) [inline docs/build-with-java.md:116]:
```suggestion
The sensor service simulates the collar's activity. You can use the following command to run the sensor service:
```

File: docs/build-with-java.md [hunk 13]
```diff
@@ -0,0 +1,261 @@
...
+
+```
+$ mvn package
+$ NODE1=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' carepet-scylla1)
+$ ./bin/sensor.sh --hosts $NODE1 --datacenter datacenter1 --measure PT1M --buffer-interval PT1M
+```
+
+The above command executes `Sensor.java` and the below main function:
```
Comment (annastuchlik) [inline docs/build-with-java.md:124]:
```suggestion
The above command executes `Sensor.java` and the following `main` function:
```

File: docs/build-with-java.md [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+```
+public static void main(String[] args) {
+        final Sensor client = new Sensor(Config.parse(new SensorConfig(), args));
+        client.save();
+        client.run();
+    }
+}
+```
+
+First, we create a client obkect which is an instance of the Sensor class. Just like in the `Migrate` class, we parse args using `Config.parse()` method to connect to the database.
```
[the remark this answers is not in the document]
Comment (annastuchlik) [reply docs/build-with-java.md:135]: A small update to Tzach's comment"
First, you create a client object - an instance of the `Sensor` class. Like in the `Migrate` class, the `Config.parse()` method parses the arguments to connect to the database.

File: docs/build-with-java.md [hunk 15]
```diff
@@ -0,0 +1,261 @@
...
+
+In the `Sensor` constructor, we attribute a random ID to the `owner`, `pet` and `sensors`.
```
Comment (annastuchlik) [inline docs/build-with-java.md:137]:
```suggestion
In the `Sensor` constructor, a random ID is attributed to the `owner`, `pet`, and `sensors`.
```

File: docs/build-with-java.md [hunk 16]
```diff
@@ -0,0 +1,261 @@
+## Build an IoT App with Java
+
+### Architecture
+
+In this section, we will walk through and explain the code for the different commands.
```
[the remark this answers is not in the document]
Comment (annastuchlik) [reply docs/build-with-java.md:5]: The correct phrase is "walk you through" and the meaning is the same as "explain how to do something", so the sentence should be: This section will walk you through the code for the different commands.
In fact, the sentence should be the same as in the section for Go.
In this section, we will walk you through the CarePet commands and explain the code behind them.

Comment (annastuchlik) [inline docs/build-with-java.md:5]: In this section, we will walk you through the CarePet commands and explain the code behind them.

**Review by annastuchlik — CHANGES_REQUESTED** [review]
I've added some comments.

**Review by annastuchlik — CHANGES_REQUESTED** [review]
passowrd -> password

**Pull request merged**